internal/clients/sources/sources_client.go

```diff
@@ -0,0 +1,153 @@
+package sources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/cache"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/headers"
+	"github.com/RHEnVision/provisioning-backend/internal/parsing"
+)
+
+type SourcesClient struct {
+	client *ClientWithResponses
+}
+
+func init() {
+	GetSourcesClientV2 = newSourcesClient
+}
+
+func newSourcesClient(ctx context.Context) (ClientV2, error) {
+	c, err := NewClientWithResponses(config.Sources.URL)
+	if err != nil {
+		return nil, err
+	}
+	return &SourcesClient{client: c}, nil
+}
+
+func (c *SourcesClient) ListProvisioningSources(ctx context.Context) (*[]Source, error) {
+	ctxval.Logger(ctx).Info().Msg("Listing provisioning sources")
+	appTypeId, err := c.GetProvisioningTypeId(ctx)
+	if err != nil {
+		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to get provisioning type id")
+		return nil, fmt.Errorf("failed to get provisioning app type: %w", err)
+	}
+	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddIdentityHeader)
+	if err != nil {
+		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to fetch ApplicationTypes from sources")
+		return nil, fmt.Errorf("failed to get ApplicationTypes: %w", err)
+	}
+	statusCode := resp.StatusCode()
+
+	if parsing.IsHTTPNotFound(statusCode) {
+		return nil, NewNotFoundError(ctx, err, "sources not found")
```

Comment:
If we have decided to use a generic NewNotFoundError
we shouldn't pass here err because it is nil, should we create a custom dynamic one?